nsis/src/nsis/tests/NsisExecutableTest.java

```diff
@@ -1,28 +1,96 @@
 package nsis.tests;
 
+import static org.junit.Assert.assertEquals;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
 import org.junit.Test;
 
 import generic.continues.RethrowContinuesFactory;
 import ghidra.app.util.bin.ByteArrayProvider;
 import ghidra.app.util.bin.format.pe.PortableExecutable.SectionLayout;
 import nsis.file.NsisExecutable;
-import static org.junit.Assert.*;
-
-import java.io.IOException;
 
 public class NsisExecutableTest {
 
-	private byte[] mockFile = { (byte) 0xef, (byte) 0xef, (byte) 0xef,
-			(byte) 0xbe, (byte) 0xad, (byte) 0xde, 'N', 'u', 'l', 'l', 's', 'o',
-			'f', 't', 'I', 'n', 's', 't' };
+	private final String pathWithoutCompression = "testData/nsis_without_compression.exe";
+	private final String pathWithBzip = "testData/nsis_with_bzip.exe";
+	private final String pathWithLZMA = "testData/nsis_with_lzma.exe";
+	private final String pathWithZlib = "testData/nsis_with_zlib.exe";
+
+	@Test
+	public void testNsisCreationNotCompressed() throws IOException {
+		InputStream binaryInputStream = new FileInputStream(
+				new File(pathWithoutCompression));
+		ByteArrayProvider bp = new ByteArrayProvider(
+				binaryInputStream.readAllBytes());
+		NsisExecutable ne = NsisExecutable.createNsisExecutable(
+				RethrowContinuesFactory.INSTANCE, bp, SectionLayout.FILE);
+
+		assertEquals((long) 36356, ne.getHeaderOffset());
```

Comment:
I would like this to be hex. If we have to debug something we will probably look at it in a hex editor and all offsets will be in hex.